mantidproject/mantid
# (OSIRIS) Diffraction Reduction - Remove Manual D-Ranges and fully automate

Currently, the OSIRISDiffractionReduction algorithm (and consequently the Diffraction Reduction interface, when OSIRIS is selected as the instrument) is able to be supplied a manual d-range, which is used to relate the sample, container and vanadium runs.

This routine however, can be fully automated by examining the TOF range of the runs workspaces and subsequently deriving their D-Range. This automation was removed with the addition of containers to the OSIRISDiffractionReduction algorithm and is reapplied by this PR.

This PR ensures that the smallest overlapping (between sample runs, container runs and vanadium runs) subset of D-Ranges provided will be used in producing a result (files not in this subset will be ignored).
e.g.
If Sample Runs of d3-d4 were provided, Container Runs of d2-d5 were provided and vanadium runs of d1-d10 were provided, the diffraction reduction algorithm will produce a result for d3-d4 only.

**To test:**
1. Navigate to Interfaces > Indirect > Diffraction.
2. Select 'OSIRIS' in the Instrument drop-down menu.
3. Enter `10203, 10205, 10204, 10206` into the 'Sample' field.
4. Enter `10241, 10242, 10244, 10243` into the 'Container' field.
5. Enter `10159, 10158, 10157, 10156` into the 'Vanadium' field.
**Note the runs are provided in varying orders - this is to test that the order is not considered, runs should be organised by d-range**
6. Provide the supplied calibration file into the 'Calibration' field.
7. Click 'Run'.

**Test Files**
[osiris_calibration_file.zip](https://github.com/mantidproject/mantid/files/1479559/osiris_calibration_file.zip)



Fixes #21192 . 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
